pkg/operator/controller/controller.go

```diff
@@ -101,64 +100,56 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 			// resource that got deleted async).
 			log.Info("clusteringress not found; reconciliation will be skipped", "request", request)
 		} else {
-			// Try again later.
-			log.Error(err, "failed to get clusteringress", "request", request)
-			result.RequeueAfter = 10 * time.Second
+			errs = append(errs, fmt.Errorf("failed to get clusteringress %q: %v", request, err))
 		}
 		ingress = nil
 	}
 
 	caSecret, err := r.ensureRouterCACertificateSecret()
 	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to ensure CA secret: %v", err)
+		errs = append(errs, fmt.Errorf("failed to ensure CA secret: %v", err))
+		caSecret = nil
 	}
 
-	// Collect errors as we go.
-	errs := []error{}
-
-	// Only reconcile the ingress itself if it exists.
 	if ingress != nil {
 		dnsConfig := &configv1.DNS{}
 		if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, dnsConfig); err != nil {
-			log.Error(err, "failed to get dns 'cluster'")
+			errs = append(errs, fmt.Errorf("failed to get dns 'cluster': %v", err))
 			dnsConfig = nil
 		}
 		infraConfig := &configv1.Infrastructure{}
 		if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
-			log.Error(err, "failed to get infrastructure 'cluster'")
+			errs = append(errs, fmt.Errorf("failed to get infrastructure 'cluster': %v", err))
 			infraConfig = nil
 		}
 		ingressConfig := &configv1.Ingress{}
 		if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, ingressConfig); err != nil {
-			log.Error(err, "failed to get ingress 'cluster'")
+			errs = append(errs, fmt.Errorf("failed to get ingress 'cluster': %v", err))
 			ingressConfig = nil
 		}
 
-		if dnsConfig == nil || infraConfig == nil || ingressConfig == nil {
-			// For now, if the cluster configs are unavailable, defer reconciliation
-			// because weaving conditionals everywhere to deal with various nil states
-			// is too complicated. It doesn't seem too risky to rely on the invariant
-			// of the cluster config being available.
-			result.RequeueAfter = 10 * time.Second
-		} else {
-			// Ensure we have all the necessary scaffolding on which to place router
-			// instances.
-			err = r.ensureRouterNamespace()
-			if err != nil {
-				errs = append(errs, err)
-				return result, utilerrors.NewAggregate(errs)
+		// For now, if the cluster configs are unavailable, defer reconciliation
+		// because weaving conditionals everywhere to deal with various nil states
+		// is too complicated. It doesn't seem too risky to rely on the invariant
+		// of the cluster config being available.
+		if dnsConfig != nil && infraConfig != nil && ingressConfig != nil {
+			// Ensure we have all the necessary scaffolding on which to place router instances.
+			if err = r.ensureRouterNamespace(); err != nil {
```

Comment:
It's an extremely minor concern, but I think `:=` would be preferable here because if there is no reason to change the variable in the outer scope, we shouldn't.